app/src/main/java/org/gdg/frisbee/android/about/AboutActivity.java

```diff
@@ -72,17 +75,26 @@ public boolean onOptionsItemSelected(MenuItem item) {
             finish();
             return true;
         } else if (R.id.action_app_invite == item.getItemId()) {
-            Intent intent = new AppInviteInvitation.IntentBuilder(getString(R.string.invitation_title))
-                .setMessage(getString(R.string.invitation_message))
-                .setDeepLink(Uri.parse(getString(R.string.invitation_deep_link)))
-                .build();
-            startActivityForResult(intent, REQUEST_INVITE);
+            shareAppInviteLink();
             return true;
         }
 
         return super.onOptionsItemSelected(item);
     }
 
+    private void shareAppInviteLink() {
+        AppInviteLinkGenerator linkGenerator = AppInviteLinkGenerator.create();
+        String gplusId = PlusUtils.getCurrentPersonId(getGoogleApiClient());
+        HttpUrl appInviteLink = gplusId != null
```

Comment:
AboutActivity should not decide on how to handle not signed users. This should go into the linkGenerator.